engine/gameobject/src/gameobject/res_collection.cpp

```diff
@@ -212,10 +213,10 @@ namespace dmGameObject
         return res;
     }
 
-    static dmResource::Result ResCollectionPreload(const dmResource::ResourcePreloadParams& params)
+    static dmResource::Result ResCollectionPreload(const dmResource::ResourcePreloadParams* params)
     {
         dmGameObjectDDF::CollectionDesc* collection_desc;
-        dmDDF::Result e = dmDDF::LoadMessage<dmGameObjectDDF::CollectionDesc>(params.m_Buffer, params.m_BufferSize, &collection_desc);
+        dmDDF::Result e = dmDDF::LoadMessage<dmGameObjectDDF::CollectionDesc>(params->m_Buffer, params->m_BufferSize, &collection_desc);
```

Comment:
After going back and forth, I kept the structs as public structs, i.e. part of the API...